src/zenml/integrations/pandas/materializers/pandas_materializer.py

```diff
@@ -204,14 +282,46 @@ def extract_metadata(
         Returns:
             The extracted metadata as a dictionary.
         """
-        pandas_metadata: Dict[str, "MetadataType"] = {"shape": df.shape}
+        # Store whether it's a Series for later reference
+        is_series = isinstance(df, pd.Series)
 
-        if isinstance(df, pd.Series):
-            pandas_metadata["dtype"] = DType(df.dtype.type)
-            pandas_metadata["mean"] = float(df.mean().item())
-            pandas_metadata["std"] = float(df.std().item())
-            pandas_metadata["min"] = float(df.min().item())
-            pandas_metadata["max"] = float(df.max().item())
+        # Store original shape before conversion
+        original_shape = df.shape
+
+        # Convert Series to DataFrame for consistent handling of dtypes
+        if is_series:
+            series_obj = df  # Keep original Series for some calculations
+            df = df.to_frame(name="series")
+
+        pandas_metadata: Dict[str, "MetadataType"] = {"shape": original_shape}
+
+        # Add information about custom data types to metadata
+        custom_types = {}
+        try:
+            for col, dtype in df.dtypes.items():
+                dtype_str = str(dtype)
+                if not is_standard_dtype(dtype_str):
+                    col_name = "series" if is_series else str(col)
+                    custom_types[col_name] = dtype_str
+                    # Try to get module information if available
+                    try:
+                        module_name = dtype.type.__module__.split(".")[0]
+                        custom_types[f"{col_name}_module"] = module_name
```

Comment:
This is trying to extract the package, not the actual module in which the type is stored, correct?

Reply:
its trying to extract the module_name

Reply:
The module name is the full `__module__` before you're splitting it and only taking the first element. And usually (not always), this first element is the PyPI package which contains the module.
So just a naming thing, if you want to call it module just take `dtype.type.__module__`